src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -2339,7 +2371,12 @@ bool QgsPostgresProvider::addFeatures( QgsFeatureList &flist, Flags flags )
 
       if ( i == flist.size() )
       {
-        if ( qgsVariantEqual( v, defVal ) )
+        // generated values override everything.
+        if ( !mGeneratedValues[idx].isEmpty() )
+        {
+          values += delim + defVal;
```

Comment:
```suggestion
          values += delim + QStringLiteral( "DEFAULT" );
```

We need to specify `DEFAULT` here, not the calculation expression, right?

Reply:
Actually this is dead code, since we don't use DEFAULT anymore, and skip the GENERATED field entirely (line 2342)